File: src/main/java/com/kingmang/ixion/ast/PostfixExpression.java
package com.kingmang.ixion.ast;

import com.kingmang.ixion.ExprVisitor;
import com.kingmang.ixion.lexer.Position;
import com.kingmang.ixion.lexer.Token;

public final class PostfixExpression extends Expression {
    public final Expression expression;
    public final Token operator;
    public int localIndex = -1;

    public PostfixExpression(Position pos, Expression expression, Token operator) {
        super(pos);
        this.expression = expression;
        this.operator = operator;
    }

    public <R> R accept(ExprVisitor<R> visitor) {
        return visitor.visitPostfixExpr(this);
    }
}

File: src/main/java/com/kingmang/ixion/ast/PostfixExpression.kt
package com.kingmang.ixion.ast

import com.kingmang.ixion.ExprVisitor
import com.kingmang.ixion.lexer.Position
import com.kingmang.ixion.lexer.Token

class PostfixExpression(pos: Position?, @JvmField val expression: Expression?, @JvmField val operator: Token?) : Expression(pos) {
    var localIndex: Int = -1

    override fun <R> accept(visitor: ExprVisitor<R>): R {
        return visitor.visitPostfixExpr(this)
    }
}